File: plsql-squid/src/test/java/br/com/felipezorzo/sonar/plsql/statements/CloseStatementTest.java
package br.com.felipezorzo.sonar.plsql.statements;

import static org.sonar.sslr.tests.Assertions.assertThat;

import org.junit.Before;
import org.junit.Test;

import br.com.felipezorzo.sonar.plsql.api.PlSqlGrammar;
import br.com.felipezorzo.sonar.plsql.api.RuleTest;

public class CloseStatementTest extends RuleTest {

    @Before
    public void init() {
        setRootRule(PlSqlGrammar.CLOSE_STATEMENT);
    }
    
    @Test
    public void matchesSimpleClose() {
        assertThat(p).matches("close foo;");
    }
    
    @Test
    public void matchesSimpleCloseHostCursor() {
        assertThat(p).matches("close :foo;");
    }
    
    @Test
    public void matchesLabeledClose() {
        assertThat(p).matches("<<foo>> close foo;");
    }

}

File: zpa-core/src/test/kotlin/com/felipebz/zpa/api/statements/CloseStatementTest.kt
/**
 * Z PL/SQL Analyzer
 * Copyright (C) 2015-2025 Felipe Zorzo
 * mailto:felipe AT felipezorzo DOT com DOT br
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.felipebz.zpa.api.statements

import com.felipebz.flr.tests.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import com.felipebz.zpa.api.PlSqlGrammar
import com.felipebz.zpa.api.RuleTest

class CloseStatementTest : RuleTest() {

    @BeforeEach
    fun init() {
        setRootRule(PlSqlGrammar.CLOSE_STATEMENT)
    }

    @Test
    fun matchesSimpleClose() {
        assertThat(p).matches("close foo;")
    }

    @Test
    fun matchesCloseCursorInPackage() {
        assertThat(p).matches("close pack.foo;")
    }

    @Test
    fun matchesSimpleCloseHostCursor() {
        assertThat(p).matches("close :foo;")
    }

    @Test
    fun matchesLabeledClose() {
        assertThat(p).matches("<<foo>> close foo;")
    }

}
